### DISCUSSION TITLE
OrientDB Enterprise Agent released as Open Source!

### DISCUSSION THREAD

**Discussion by luigidellaquila** [orientechnologies/orientdb#9756] [General]
Hi all,

I have an important announcement to share today: SAP decided to release OrientDB Enterprise Agent as Open Source, with the exact same license as OrientDB Community Edition, i.e. Apache 2.0.

Here you can find the [official announcement](https://orientdb.org/enterprise-agent-open-source) (intended for the customer base mostly) and here you can find some of [my very personal views](https://luigidellaquila.github.io/orientdb/release/oss/2022/01/25/orientdb-enterprise-oss.html) on this topic.

IMHO it's a huge opportunity, especially for the existing user base, that can now benefit from some really interesting features (as for OrientDB Community Edition, for free!), but also for the future of the project as a whole.

A public binary release will follow very soon, so stay tuned ;-)

Thanks!

Luigi

**Comment by lvca** [comment]
This was the right decision to take a long time ago! Anyway, better late than never, right?

Thanks, @luigidellaquila for your last contribution to the OrientDB project before leaving SAP. I hope clients can safely adopt this new open-source Enterprise Edition module or perhaps switch to [ArcadeDB](https://github.com/ArcadeData/arcadedb).

**Reply by smolinari** [reply]
Interesting. I guess "on prem" is dead for SAP. Very interesting. Are there plans with SAP to have a cloud offering of OrientDB? If not, and even when, sounds like a perfect door opening for someone to fill that gap. 

Scott

**Comment by adriancoca** [comment]
@luigidellaquila where I can find that "public binary release" ?

**Reply by tglman** [reply]
hi, you can fined a public binary release on the maven central: https://search.maven.org/artifact/com.orientechnologies/agent

**Comment by gsbomrel** [comment]
@luigidellaquila, @lvca 
Does this mean we can use orient db enterprise version other than 3.0 like 2.2.37 in production server ?

**Reply by tglman** [reply]
version 2.2.x are out of support and no new release  will be shipped, is strongly advised to migrate to a 3.x  release, suggested a 3.2.x release

**Reply by Eric24** [reply]
Understood that 2.2.x is quite old, but we still have a few of these running. It would be very helpful to use the Enterprise Agent to help with the migration. What is the last version of the Agent that will still work with a 2.2.x database (and where can I get it)?

**Reply by tglman** [reply]
Back in time the enterprise agent was not open source, so I do not think I can really provide a version of it for 2.2.x even if I was technically able to do that, my suggestion is really to plan a migration to 3.2.x, that is way more stable than 2.2.x and I did also get some reports from users of performance improvements in their use cases after upgrade, so there is more than one reason to upgrade, also the upgrade should be pretty smooth, you can contact me if you have problem with it.

**Reply by Eric24** [reply]
@tglman - Thank you for the information. I understand the potential licensing issues (I actually think it would be OK under the circumstances, but I suppose there's no one to ask). In any case, my objective is to upgrade to 3.2.x (do you recommend a particular release?), and the use of the Enterprise Agent (particular Teleporter) was part of my plan to do that (I need not only to upgrade, but also to split into two separate server groups). Is there a way to use Enterprise Agent (and Teleporter) under 3.2.x and connect to a 2.2.x server as a data source? Outside of that, is there documentation anywhere that describes the recommended approach to upgrading?

**Reply by Eric24** [reply]
@tglman - PS: I'm using the OrientJS driver on node.js on the client side. I'm interested in understanding any breaking changes on that front, too.

**Reply by tglman** [reply]
Hi,

For upgrading it should be pretty smooth, we do suggest to do a [database export](https://orientdb.dev/docs/3.2.x/console/Console-Command-Export.html)from the 2.2.x, and do a  [database import](https://orientdb.dev/docs/3.2.x/console/Console-Command-Import.html)  in 3.2.x to make sure that you start with more recent datastructures.

orientjs driver here: https://github.com/orientechnologies/orientjs is tested with 3.2.x versions

**Reply by Eric24** [reply]
OK, two questions:

1. For 3.2.x, is there any reason not to use the latest (3.2.32)?
2. You say the export/import is recommended. So assuming it's not mandatory, what's the risk of using the same 2.2.x database with 3.2.x?

**Reply by tglman** [reply]
It should be possible to migrate the database directly without export and import, it may require though to go trough all the 3.x versions, to make sure automatic migrations are run, even if that is done and it works correctly, the databases will still be using some old data structures, that may not be as performant as the most recent, so is strongly suggested to do and export+import

**Comment by MarkusNetz** [comment]
Will the agent always be a separate module or will it eventually be included in the tar.gz package from main page?
As of now I will have to put a maven dependency.
At my place I download the tar.gz and then make an RPM package of it to release internally.
Now I plan on getting the .jar included in the RPM packaging so it ends up in the correct installation dir.

**Reply by andrii0lomakin** [reply]
Hi, we need to discuss this. Then I will be back to you.

**Comment by alxxyz** [comment]
Does it mean that we can use it for free?

**Reply by andrii0lomakin** [reply]
Yes

**Reply by sblommers** [reply]
I noticed a lot of packages indicating data sharing to public cloud (ex. AWS). What kind of information sharing is the agent responsible for? We decided to run on-premise without any open connection to the outside world with 2.2.x, will that be a limitation for using the agent with the fresh 3.2.6?

**Reply by tglman** [reply]
OrientDB does not share data back or upload any data to any public cloud, there are features that allow for example to upload your backups on public clouds, but you have to configure that yourself with your own credentials, you can still run OrientDB on-premise and off the internet.

**Comment by timw** [comment]
@tglman / @luigidellaquila  - Would the maintainers be receptive to PRs incorporating some of the enterprise plugin functionality, or do you already have plans for that?

**Reply by tglman** [reply]
Hi,

For I while we would prefer to keep the functionality provided by the EE agent usable only trough the EE agent, for non technical reason, this may be not need to be the case anymore let's say in two year time.

Regards

**Reply by adriancoca** [reply]
@tglman  any plans to include EE agent in ORIENT release? It would be easier if is included by default in docker orient image.

**Reply by tglman** [reply]
hi @adriancoca,

So for the 3.2.x series there is no plan to include part for the EE agent in the base release, but there are plans to merge those features in the main repo, so the next major have them included, even though there is no timiline/deadline for that
